experimental/dds/tree2/src/core/rebase/utils.ts

```diff
@@ -37,7 +38,14 @@ export function rebaseBranch<TChange>(
 	changeRebaser: ChangeRebaser<TChange>,
 	sourceHead: GraphCommit<TChange>,
 	targetHead: GraphCommit<TChange>,
-): [newSource: GraphCommit<TChange>, sourceChange: TChange];
+): [
+	newSourceHead: GraphCommit<TChange>,
+	sourceChange: TChange | undefined,
+	commits: {
+		deletedSourceCommits: GraphCommit<TChange>[];
```

Comment:
What are deleted commits?

Reply:
Yea, admittedly, I didn't really doc these because I have a feeling they might change a bit more in upcoming refactors. But, I should doc them anyway. I factored out this `commits` type and documented each field.